src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -43,6 +57,7 @@ public static TraceListenerCollection Listeners
                             defaultListener.IndentLevel = t_indentLevel;
                             defaultListener.IndentSize = s_indentSize;
                             s_listeners.Add(defaultListener);
+                            Debug.SetProvider(s_debugProvider);
```

Comment:
Need a comment here explaining why you're setting it here. It's because this is the first point where you know for sure you have some listeners?

Reply:
correct. OK, I'll add the comment